app/assets/javascripts/listings/ListingDataService.js.coffee

```diff
@@ -0,0 +1,434 @@
+        i = 0
+        amiPercentChart.values.forEach (incomeLevel) ->
+          chartAmount = if chart.values[i] then chart.values[i].amount else 0
+          incomeLevel.amount = Math.max(incomeLevel.amount, chartAmount)
+          i++
+    charts
+
+  Service.priorityTypes = (listing) ->
+    Service.collectTypes(listing, 'prioritiesDescriptor')
+
+  Service.collectTypes = (listing, specialType) ->
+    _.map listing[specialType], (descriptor) ->
+      descriptor.name
+
+  Service.priorityLabel = (priority, modifier) ->
+    labelMap =
+      'Vision impairments':
+        name: 'Vision Impairments'
+        description: 'impaired vision'
+      'Hearing impairments':
+        name: 'Hearing Impairments'
+        description: 'impaired hearing'
+      'Hearing/Vision impairments':
+        name: 'Vision and/or Hearing Impairments'
+        description: 'impaired vision and/or hearing'
+      'Mobility/hearing/vision impairments':
+        name: 'Mobility, Hearing and/or Vision Impairments'
+        description: 'impaired mobility, hearing and/or vision'
+      'Mobility impairments':
+        name: 'Mobility Impairments'
+        description: 'impaired mobility'
+
+    return priority unless labelMap[priority]
+    return labelMap[priority][modifier]
+
+  Service.reservedLabel = (listing, type,  modifier) ->
+    labelMap =
+      "#{ListingConstantsService.RESERVED_TYPES.SENIOR}":
+        building: 'Senior'
+        eligibility: 'Seniors'
+        reservedFor: "seniors #{Service.seniorMinimumAge(listing)}"
+        reservedForWhoAre: "seniors #{Service.seniorMinimumAge(listing)}"
+        unitDescription: "seniors #{Service.seniorMinimumAge(listing)}"
+      "#{ListingConstantsService.RESERVED_TYPES.VETERAN}":
+        building: 'Veterans'
+        eligibility: 'Veterans'
+        reservedFor: 'veterans'
+        reservedForWhoAre: 'veterans'
+        unitDescription: 'veterans of the U.S. Armed Forces'
+      "#{ListingConstantsService.RESERVED_TYPES.DISABLED}":
+        building: 'Developmental Disability'
+        eligibility: 'People with developmental disabilities'
+        reservedFor: 'people with developmental disabilities'
+        reservedForWhoAre: 'developmentally disabled'
+        unitDescription: 'people with developmental disabilities'
+
+    return type unless labelMap[type]
+    return labelMap[type][modifier]
+
+  # used by My Applications -- when you load an application we also parse the attached listing data
+  Service.loadListing = (listing) ->
+    return if Service.listing && Service.listing.Id == listing.Id
+    # TODO: won't be needed if we ever consolidate Listing_Lottery_Preferences and /preferences API
+    listing.preferences = _.map listing.Listing_Lottery_Preferences, (lotteryPref) ->
+      {
+        listingPreferenceID: lotteryPref.Id
+        preferenceName: lotteryPref.Lottery_Preference.Name
+      }
+    angular.copy(listing, Service.listing)
+
+  Service.seniorMinimumAge = (listing) ->
```

Comment:
I'm not sure this warrants its own function. Should it just move up to reservedLabel? otherwise it's not clear from the function name that it's actually returning a formatted string with '+' in it, rather than a number.

Reply:
we can make it inline in reservedLabel like `listing.Reserved_community_minimum_age ? "#{listing.Reserved_community_minimum_age}+" : ''` 
but it's a bit long and going along with DRY spirit I would rename it to sth like `formatSeniorMinAge`. What do you think?

Reply:
sure that works for me. Let's move it up so it's right next to reservedLabel too if possible